### DISCUSSION TITLE
I'm not abe to commit my changes, can anyone pls help me to fix this

### DISCUSSION THREAD

**Discussion by mr-sahu2002** [TheAlgorithms/Python#10577] [Q&A]
![Screenshot 2023-10-16 105758](https://github.com/TheAlgorithms/Python/assets/70091452/28415d2a-61c9-4d24-beac-ed5468678579)
![Screenshot 2023-10-16 114753](https://github.com/TheAlgorithms/Python/assets/70091452/27cb6261-aee7-4243-8b74-c4672a926b24)

**Comment by sight09** [comment]
Hi there! Your commit is being blocked by local pre-commit hooks because a few code styling and formatting checks failed.

Here is exactly what failed and how to fix it:
1. End of File Fixer (end-of-file-fixer)

    What happened: Python files require a single blank newline at the very end of the file. The hook automatically modified Automation/auto_type.py to add this for you.

    The Fix: Because the hook modified the file on your disk, the changes are now unstaged. You just need to re-stage the file:
    Bash

    git add Automation/auto_type.py

2. Ruff Import Sorting (ruff)

    What happened: The linter noticed that your imports in auto_type.py are out of order or unformatted (I001 Import block is un-sorted).

    The Fix: You can let Ruff fix this automatically by running:
    Bash

    pre-commit run ruff --all-files

    (Alternatively, you can just manually move import keyboard above import time to alphabetize them, or use ruff check --fix).

3. Validate Filenames (validate-filenames)

    What happened: The repository has a strict naming convention for files and folders. Your folder name Automation/ starts with a capital A, which is likely breaking a lowercase-only rule.

    The Fix: Rename your directory to use all lowercase letters:
    Bash

    # Rename the directory to lowercase
    mv Automation automation

    # Stage the newly renamed directory
    git add automation/

🚀 Final Steps to Commit

Once you have renamed the folder, fixed the imports, and staged the changes, run your commit command again:
Bash

git add .
git commit -m "new directory automation"

Everything should show green Passed now! Let me know if this works for you.
